router/transformer/transformer.go

```diff
@@ -48,8 +49,23 @@ type handle struct {
 	logger                    logger.Logger
 }
 
+type ProxyRequestMetadata struct {
+	JobID         int64           `json:"jobId"`
+	AttemptNum    int             `json:"attemptNum"`
+	UserID        string          `json:"userId"`
+	SourceID      string          `json:"sourceId"`
+	DestinationID string          `json:"destinationId"`
+	WorkspaceID   string          `json:"workspaceId"`
+	Secret        json.RawMessage `json:"secret"`
+	DestInfo      json.RawMessage `json:"destInfo,omitempty"`
+}
+
+type ProxyRequestPayload struct {
+	*integrations.PostParametersT
```

Comment:
Why are we embedding a pointer instead of a value? We don't want to treat nils, do we?

Reply:
Basically I followed a stackoverflow thread where in combining two structs to send it as aa json was shown in this way! 😓 

```go
type A struct {
a string
a1 string
}

type B struct {
b string
}

type C struct {
 A,
 B
}

/*
Can we achieve this ?
output json should look like this
{
 a: "v1",
 a1: "v11",
 b: "v2"
}
*/
```

Reply:
embedding is fine, embedding a value instead of a pointer is preferred here
```go
type ProxyRequestPayload struct {
	integrations.PostParametersT
	Metadata ProxyRequestMetadata `json:"metadata,omitempty"`
}
```